pkg/csi/service/common/authmanager.go

```diff
@@ -135,44 +184,244 @@ func GenerateDatastoreIgnoreMapForBlockVolumes(ctx context.Context, vc *vsphere.
+	log := logger.GetLogger(ctx)
+	log.Debugf("Computing the cluster to file service status (enabled/disabled) map.")
+	datacenters, err := vc.ListDatacenters(ctx)
+	if err != nil {
+		log.Errorf("failed to find datacenters from VC: %+v, Error: %+v", vc.Config.Host, err)
+		return nil, err
+	}
+
+	dsToFileServiceEnabledMap := make(map[string]bool)
+	// Get clusters from datacenters
+	clusterComputeResources := []*object.ClusterComputeResource{}
+	for _, datacenter := range datacenters {
+		finder := find.NewFinder(datacenter.Datacenter.Client(), false)
+		finder.SetDatacenter(datacenter.Datacenter)
+		clusterComputeResource, err := finder.ClusterComputeResourceList(ctx, "*")
+		if err != nil {
+			if _, ok := err.(*find.NotFoundError); ok {
+				log.Debugf("No clusterComputeResource found in dc: %+v. error: %+v", datacenter, err)
+				continue
+			}
+			log.Errorf("Error occurred while getting clusterComputeResource. error: %+v", err)
+			return nil, err
+		}
+		clusterComputeResources = append(clusterComputeResources, clusterComputeResource...)
+	}
+
+	// Get Clusters with HostConfigStoragePriv
+	authMgr := object.NewAuthorizationManager(vc.Client.Client)
+	privIds := []string{HostConfigStoragePriv}
+	userName := vc.Config.Username
+	var entities []types.ManagedObjectReference
+	clusterComputeResourcesMap := make(map[string]*object.ClusterComputeResource)
+	for _, cluster := range clusterComputeResources {
+		entities = append(entities, cluster.Reference())
+		clusterComputeResourcesMap[cluster.Reference().Value] = cluster
+	}
+
+	// invoke authMgr function HasUserPrivilegeOnEntities
+	result, err := authMgr.HasUserPrivilegeOnEntities(ctx, entities, userName, privIds)
+	if err != nil {
+		log.Errorf("auth manager: failed to check privilege %v on entities %v for user %s", privIds, entities, userName)
+		return nil, err
+	}
+	log.Debugf("auth manager: HasUserPrivilegeOnEntities returns %v when checking privileges %v on entities %v for user %s", result, privIds, entities, userName)
+	clusterComputeResourceWithPriv := []*object.ClusterComputeResource{}
+	for _, entityPriv := range result {
+		hasPriv := true
+		privAvails := entityPriv.PrivAvailability
+		for _, privAvail := range privAvails {
+			// required privilege is not grant for this entity
+			if !privAvail.IsGranted {
+				hasPriv = false
+				break
+			}
+		}
+		if hasPriv {
+			clusterComputeResourceWithPriv = append(clusterComputeResourceWithPriv, clusterComputeResourcesMap[entityPriv.Entity.Value])
+		}
+	}
+	log.Debugf("Clusters with priv: %s are : %+v", HostConfigStoragePriv, clusterComputeResourceWithPriv)
+
+	// Get clusters which are vSAN and have vSAN FS enabled
+	pc := property.DefaultCollector(vc.Client.Client)
+	properties := []string{"info", "summary"}
+	// Get all the vsan datastores with vsan FS from these clusters and add it to map.
+	for _, cluster := range clusterComputeResourceWithPriv {
```

Comment:
I dont think this code would work in case of VMC (assuming vSAN file services is supported). Basically, in this code we are assuming that all vSAN datastores are candidates under the VC cluster that has file services enabled. In VMC we have management and workload vSAN datastores and only workload datastores are eligible for file volume provisioning.

Reply:
>Basically, in this code we are assuming that all vSAN datastores are candidates under the VC cluster that has file services enabled.

For clusters which have `Host.Config.Storage` permission on vSAN clusters with vSAN FS enabled. Then we get vSAN datastores from these clusters. Later we check `Datastore.FileManagement` and `System.Read` permission on these vSAN datastores. Only those datastores would be picked for file volume provisioning. 

>In VMC we have management and workload vSAN datastores and only workload datastores are eligible for file volume provisioning.

I hope this question is answered with above explanation.

Reply:
> Later we check Datastore.FileManagement and System.Read permission on these vSAN datastores. Only those datastores would be picked for file volume provisioning

These privs are not required for file volume provisioning.

Reply:
My mistake, you are correct! I tested this scenario js now and it works without permission. I will make the change accordingly.

Reply:
>In VMC we have management and workload vSAN datastores and only workload datastores are eligible for file volume provisioning.

 But didn't get this, how do we deal with this? Do we need additional priv checking for this?

Reply:
No, we need to know which vSAN datastore is management and which one is a workload datastore. For now, can we put a TODO so that we know this code needs improvement to support file volumes in VMC?